core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -727,4 +729,21 @@ public boolean accept(File userStores, String name) {
     private void clearTenantCache(int tenantId) {
         tenantCacheManager.clearCacheEntry(new TenantIdKey(tenantId));
     }
+
+    private void deleteCachemanager(String tenantDomain, int tenantId) {
+
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            carbonContext.setTenantDomain(tenantDomain);
+            carbonContext.setTenantId(tenantId);
+
+            Caching.getCacheManagerFactory().close();
+        } catch (Exception e) {
+            log.error("Unable to delete cache for tenant :"+tenantDomain);
```

Comment:
log.error("Unable to delete cache for tenant :"+tenantDomain, e);

Prints the stacktrace upon error, so that finding root cause is easy.